File: scripts/refresh-players.js
const fs = require('fs');
const path = require('path');

const API_URL = 'https://api.sleeper.app/v1/players/nfl';
const OUTPUT_PATH = path.resolve(__dirname, '../SleeperAPI.json');

async function main() {
  const response = await fetch(API_URL, {
    headers: {
      Accept: 'application/json'
    }
  });

  if (!response.ok) {
    throw new Error(`Sleeper API request failed: ${response.status} ${response.statusText}`);
  }

  const payload = await response.json();
  fs.writeFileSync(OUTPUT_PATH, `${JSON.stringify(payload, null, 2)}\n`);

  const totalPlayers = Object.keys(payload || {}).length;
  console.log(`Saved ${totalPlayers} players to ${OUTPUT_PATH}`);
}

main().catch((error) => {
  console.error('Failed to refresh player data:', error);
  process.exit(1);
});

File: scripts/refresh_players.py
#!/usr/bin/env python3
import json
import urllib.request
from pathlib import Path

API_URL = "https://api.sleeper.app/v1/players/nfl"
OUTPUT_PATH = Path(__file__).resolve().parent.parent / "SleeperAPI.json"


def main() -> None:
    with urllib.request.urlopen(API_URL, timeout=30) as response:
        payload = json.load(response)

    OUTPUT_PATH.write_text(json.dumps(payload, indent=2) + "\n", encoding="utf-8")
    player_count = len(payload) if isinstance(payload, dict) else 0
    print(f"Saved {player_count} players to {OUTPUT_PATH}")


if __name__ == "__main__":
    main()
